sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -341,6 +341,78 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
       .analyze
     comparePlans(optimized, expected)
   }
+
+  test("Nested field pruning for Aggregate") {
+    val query1 = contact.groupBy($"id")(first($"name.first").as("first")).analyze
```

Comment:
Could you add tests for the cases, `contact.(limit/repartition/sample).groupBy()...`? IIUC this PR could support them, too?

Reply:
ok, let me add some tests.